app/code/Magento/InventorySales/Plugin/StockState/CheckQuoteItemQtyPlugin.php

```diff
@@ -115,7 +115,8 @@ public function aroundCheckQuoteItemQty(
         $origQty,
         $scopeId = null
     ) {
-        $result = $this->objectFactory->create();
+        $result = $proceed($productId, $itemQty, $qtyToCheck, $origQty, $scopeId = null);
```

Comment:
This around plugin was originally intended to completely override the method's behavior. Adding a `proceed` right at the start basically makes it an after plugin now.

Reply:
Anyway, currently it looks like more appropriate solution in this case would be to add backorder information in this method, otherwise, in multi-stock environment calculations are likely to be incorrect due to usage of legacy API.

Reply:
Take a look at #2452. This fixes this issue without calling the legacy functions. It returns the backordered quantity directly from the Inventory Sales module. Needs checking/reviewing thoroughly as MSI isn't my area.

Reply:
Thanks @gwharton 

Let's continue in #2452